File: lib/models/rental.dart
import 'item.dart';

class Rental {
  final Item item;
  final DateTime startDate;
  final DateTime endDate;

  Rental(this.item, this.startDate, this.endDate);

  bool isActive() {
    final now = DateTime.now();
    return endDate.isAfter(now);
  }

  bool overlaps(DateTime start, DateTime end) {
    return start.isBefore(endDate) && end.isAfter(startDate);
  }
}

File: lib/models/rental.rb
require 'date'

module Models
  class Rental
    attr_reader :item, :start_date, :end_date

    def initialize(item, start_date, end_date)
      @item = item
      @start_date = start_date
      @end_date = end_date
    end

    def active?
      @end_date >= Date.today
    end

    def overlaps?(start_date, end_date)
      (@start_date <= end_date) && (start_date <= @end_date)
    end
  end
end
